pkg/apiserver/rest/dualwriter.go

```diff
@@ -229,10 +255,9 @@ func (d *DualWriter) Update(ctx context.Context, name string, objInfo rest.Updat
 			upstream: objInfo,
 			updated:  obj,
 		}, createValidation, updateValidation, forceAllowCreate, options)
-	}
 
-	if CurrentMode == Mode4 {
-		old, err := d.Get(ctx, name, &metav1.GetOptions{})
+	case Mode4:
+		old, err := d.Storage.Get(ctx, name, &metav1.GetOptions{})
```

Comment:
in mode 4 we should basically just pass through to d.Storage

Reply:
Nice, I'll check where else we can pass through directly for legacy and Storage.

[c833fa1](https://github.com/grafana/grafana/pull/84605/commits/c833fa1bf1f95624fbe594dbf08333677fc0ae8d)